### DISCUSSION TITLE
Any reason for react 18 as peer dependency?

### DISCUSSION THREAD

**Discussion by horseeyephil** [davidjerleke/embla-carousel#391] [Help] [answered]
Is there any particular compatibility reason why embla-carousel-react requires React 18 as a peer?
Could it have peer options like:
    "react": ^16.0.0 || ^17.0.1 || ^18.0.0"?

It would be nice to have it compatible with React 17 without having to install with --legacy-peer-deps.

Thanks for the help!

**Comment by davidjerleke** [comment]
Hi @horseeyephil,

Thank you for your question.

> Is there any particular compatibility reason why embla-carousel-react requires React 18 as a peer?
Could it have peer options like:
"react": ^16.0.0 || ^17.0.1 || ^18.0.0"?

The answer is no, no compatibility reasons. Embla works just fine with React 17. About the solution you’re suggesting: I can’t see that it’s mentioned in the [NPM documentation](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#peerdependencies)? So I don’t know if it works in all cases like with NPM and Yarn etc. 

We also have [this StackOverflow thread](https://stackoverflow.com/questions/47309598/make-your-npm-package-support-multiple-versions-of-peer-dependency) with different solutions but they aren’t mentioned in the NPM documentation neither.

Would you mind doing some investigating and testing to see if your suggestion works with both NPM, Yarn, React 18 and 17 without problems? If yes, I would be happy to accept a Pull Request and I can re-open this issue:

- #362

Best,
David

**Comment by horseeyephil** [comment]
Hi @davidjerleke !
Thanks for the quick response, and all the amazing work on Embla!
Great to clear that up - that there is no breaking compatibility issue between the versions.

I will think about the best way to test and validate according to your suggestion (I'm thinking symlinks? 🤔 )

In the meantime I'll post some notes:
It's not mentioned specifically in the `peerDependencies` section,  but I think it may be implicit that peer dependencies accept the same semver "ranges" as `dependencies` do:
https://docs.npmjs.com/cli/v8/configuring-npm/package-json#dependencies
https://docs.npmjs.com/cli/v6/using-npm/semver#ranges

I think this blog makes it pretty clear:
https://fathomtech.io/blog/understanding-peer-dependencies-in-npm/#:~:text=When%20specifying%20the%20allowed%20versions%20of%20a%20package%20in%20a%20peer%20dependency%20in%20our%20package.json%2C%20we%20often%20want%20to%20specify%20more%20liberal%20version%20ranges%20than%20we%20typically%20use%20for%20normal%20or%20dev%20dependencies.

Thanks again! Will try to get the PR up soon

**Reply by davidjerleke** [reply]
Hi again @horseeyephil,

> but I think it may be implicit that peer dependencies accept the same semver "ranges" as dependencies do:
https://docs.npmjs.com/cli/v8/configuring-npm/package-json#dependencies
https://docs.npmjs.com/cli/v6/using-npm/semver#ranges

I see. Well yes, this seems to be the case and a legit syntax.  Thanks for the links. I guess we don’t need to test it then 👍🏻.

The `embla-carousel-react` package is based on hooks so I guess we need to set the lowest react version to 16.8 in `peerDependencies`. If you don’t mind, you can try it out with 16.8 using codesandbox.io or similar. If it works as expected, we could go with this:

"react": ^16.8.0 || ^17.0.1 || ^18.0.0"

…right?

I would happily accept a PR. Thanks for creating this discussion.

Best,
David

**Comment by horseeyephil** [comment, marked as answer]
Thanks @davidjerleke !
I settled on a validation method with the help of **gitpkg**!
Here is the PR: https://github.com/davidjerleke/embla-carousel/pull/392

Best,
horseeye
